### ISSUE TITLE
Crash in -(void)dequeueWrite:(NSData*)data withCode:(JFROpCode)code

### ISSUE THREAD

**Issue by nicolasbigot** [acmacalister/jetfire#10]
I'm experiencing a crash in `-(void)dequeueWrite:(NSData*)data withCode:(JFROpCode)code`.

This crashes on line line 656 for JFRWebSocket.m: `NSInteger len = [self.outputStream write:([frame bytes]+total) maxLength:(NSInteger)(offset-total)];` with EXC_BREAKPOINT (code=1).

I'm using the latest version of your code (branch master), iOS 8.3 on device, iPhone 6.
The connexion in made on a secure websocket (wss:// protocol).

Here are the variable states during the call:

    total   uint64_t	0
    offset  uint64_t	65
    frame   NSConcreteMutableData *	91 bytes
    len     NSInteger	4341676604

This is very random, I experienced it several times but did not manage to re-create it on demand.
Hope this helps..Thanks.
Also, I'm not messing with thread, I send all JFRWebSocket messages from the main thread.

**Comment by nicolasbigot** [comment]
This might be related to [this](https://github.com/daltoniam/Starscream/issues/55).

**Comment by daltoniam** [comment]
Alright thanks for the report. I just did a commit, let me know if that seems to fix it.

**Comment by daltoniam** [comment]
Where you able to check if the latest code resolves this?

**Comment by jtreanor** [comment]
Also may be related to #9

**Comment by jtreanor** [comment]
The more I look into this the more I suspect that this is the same issue I was having in #9. Although I mitigated the issue, the crash has surfaced again, although it is much rarer now.

The fix looks good for `dequeueWrite:(NSData*)data withCode:`. I am receiving very similar crash reports for `initStreamsWithData:port:`. I have opened a PR (#11), to suggest a similar safety fix for this. Let me know what you think.
